### PR TITLE
GH-92584: Remove Installing Python Modules (Distutils version)

### PR DESCRIPTION
**Pull request by AA-Turner**
This removes ``install/index``, per the notice in the header:

> The entire ``distutils`` package has been deprecated and will be removed in Python 3.12. This documentation is retained as a reference only, and will be removed with the package.

See #108018 to retain the install scheme content in the ``sysconfig`` docs.

A

* Issue: gh-92584

----
:books: Documentation preview :books:: https://cpython-previews--108020.org.readthedocs.build/

### CODE REVIEW

**Comment by vstinner** [comment]
The doc fails to build:

```
Doc/library/site.rst:188: WARNING: undefined label: inst-alt-install-user (if the link has no caption the label must precede a section header)
Doc/using/cmdline.rst:812: WARNING: undefined label: inst-alt-install-user (if the link has no caption the label must precede a section header)
```

**Review by vstinner — APPROVED** [review]
LGTM

File: Doc/using/cmdline.rst [hunk 1]
```diff
@@ -811,7 +811,7 @@ conflict.
 
    Defines the :data:`user base directory <site.USER_BASE>`, which is used to
    compute the path of the :data:`user site-packages directory <site.USER_SITE>`
-   and :ref:`Distutils installation paths <inst-alt-install-user>` for
+   and Distutils installation paths for
```
Comment (vstinner) [inline Doc/using/cmdline.rst]: Is it really useful to keep a reference to removed Distutils? setup.py is legacy, no?

Comment (AA-Turner) [reply Doc/using/cmdline.rst]: #108040 removes the reference to Distutils & replaces setup.py with pip, I wanted to keep each PR distinct. When the first one is merged I'll resolve the conflicts.

Comment (AA-Turner) [reply Doc/using/cmdline.rst]: Conflicts resolved.

**Comment by vstinner** [comment]
A Docs job failed:

```
Doc/using/cmdline.rst:812: WARNING: undefined label: inst-alt-install-user (if the link has no caption the label must precede a section header)
```

**Comment by AA-Turner** [comment]
> A Docs job failed:

Oh, I messed up the merge. Pushed a fix.

**Pull request merged**